### PR TITLE
feat(ffe-tabs-react): Add aria-selected to tabs

### PR DESCRIPTION
**Pull request by exoen**
Fixes issue #614

### CODE REVIEW

**Comment by simjes** [comment]
Isn't `aria-selected` more appropriate for tabs?

**Comment by hdaljord** [comment]
> Isn't `aria-selected` more appropriate for tabs?

Yes, aria-selected is correct. See https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html

**Review by simjes — CHANGES_REQUESTED** [review]
Change to `aria-selected`

**Comment by ivarni** [comment]
Note that this now fails the build on an a11y lint-rule
```
error  The attribute aria-selected is not supported by the role button. This role is implicit on the element button  jsx-a11y/role-supports-aria-props
```
See [rule description](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/role-supports-aria-props.md)

Looking at the error message and [this example](https://www.stefanjudis.com/blog/aria-selected-and-when-to-use-it/#the-actual-use-case-of-code-aria-selected-code-) it seems to me that you should also add the `role="tab"` attribute before using `aria-selected`. This is consistent with the link @hdaljord provided, where the example is also using that role on the tab-button.

image

So I guess yay for linters.

**Comment by ivarni** [comment]
To follow up on that, we really should require an `aria-controls` prop in the `propType` for the buttons to make sure consumers also provide which tab it controls. There's little use telling a screenreader that this controls a tab without also telling them **which** tab it controls.

**Comment by exoen** [comment]
Yay for linters!

Thanks guys, I'll push an update soon. 

@ivarni ,  should `aria-controls` be created as a seperate issue?

**Comment by ivarni** [comment]
@exoen That's up to you really, it's just a suggestion from me. By making it a required prop we're forcing the developer to at least consider providing it but it's not really a part of the tab component itself since it would be referring to some other element in another part of the DOM. 

If you want to keep the scope down on this PR, I'm sure creating an issue is just fine. Maybe we should have a quick look at the bigger picture here and update the documentation with a full-fledged example of how to use `ffe-tabs-react` buttons with actual tabs instead of the [rather bare-bones example](https://design.sparebank1.no/styleguidist/index.html#!/Fanekomponenter).

Either way, this PR is an improvement once the `tab` role is in place so, `aria-controls` was just a digression really.

**Comment by exoen** [comment]
Then I want to limit this PR to fix #614 as it was intended and rather create new issues for `aria-controls` and documentation with examples. 

Any tips on tools and screen readers to use when developing to actually test and improve the experience when using screen readers?

**Comment by ivarni** [comment]
@exoen I have previously installed and tried to use NVDA which is free but it's not a simple task to learn to use it unfortunately, so I mostly rely on the react-a11y linter, reading about usability and trying to best implement the (sometimes conflicting) advice they give. 

There's also browser plugins like Wave and Axe and Chrome devtools have the Audit tab which can check for Accessibility but to actual test how the experience is for someone with a screen reader is something I find incredible difficult. None of those tools would have picked up one the fact that we were missing `aria-selected` and the `tab` role because there's no way they could differentiate between a regular button and a button being used as a tab control.

TL;DR: It's very difficult, unfortunately.

**Review by ivarni — COMMENTED** [review]
👍

**Review by simjes — APPROVED** [review]

**Review by ivarni — APPROVED** [review]
